csvy/validators/table_schema.py

```diff
@@ -0,0 +1,454 @@
+            array, object).
+        pattern: A regular expression pattern that the value must match. Applies to
+            types: string.
+        enum: A list of possible values for the field.
+
+    """
+
+    required: bool | None = Field(None)
+    unique: bool | None = Field(None)
+    minimum: int | float | None = Field(None)
+    maximum: int | float | None = Field(None)
+    minLength: int | None = Field(None)
+    maxLength: int | None = Field(None)
+    pattern: str | None = Field(None)
+    enum: list[Any] | None = Field(None)
+
+
+class CommonValidator(BaseModel):
+    """Validator for the columns in the Table Schema.
+
+    This class is used to validate the columns in the Table Schema. It is based on the
+    columns defined in the Table Schema specification.
+
+    Attributes:
+        name: The name of the column.
+        title: A nicer human readable label or title for the field.
+        example: An example value for the field.
+        description: A description for the field.
+        constraints: A dictionary of constraints for the field.
+
+    """
+
+    name: str = Field(..., description="Column name.")
+    title: str | None = Field(
+        None, description="A nicer human readable label or title for the field."
+    )
+    example: str | None = Field(None, description="An example value for the field.")
+    description: str | None = Field(None, description="A description for the field.")
+    constraints: ConstraintsValidator = Field(
+        ConstraintsValidator(), description="A dictionary of constraints for the field."
+    )
+
+    def model_dump(self, *args, **kwargs) -> dict[str, Any]:
+        """Dump the model to a dictionary.
+
+        This method dumps the model to a dictionary. It sets exclude_unset to True and
+        by_alias to True, so that only the attributes that were set are included in the
+        dictionary and their aliases are always used.
+
+        Finally, it converts the attributes that are Enum instances to their values.
+
+        Returns:
+            A dictionary with the model attributes.
+
+        """
+        kwargs["exclude_unset"] = True
+        kwargs["by_alias"] = True
+        output = super().model_dump(*args, **kwargs)
+        for key, value in output.items():
+            if isinstance(value, Enum):
+                output[key] = value.value
+        if "constraints" in output and not output["constraints"]:
+            del output["constraints"]
+        return output
+
+
+class DefaultColumnValidator(CommonValidator):
+    """Default column validator for anything that does not have a specific one.
+
+    Attributes:
+        type_: A string specifying the type, valid values are any of the TypeEnum that
+            do not have a specific validator.
+        format_: A string specifying a format, with valid values being 'default' and
+            None.
+
+    """
+
+    type_: Literal[
```

Comment:
I'm guessing the underscore is just because you can't have a field named `type` in Python? If so, I suppose you could also call it `kind` instead

Reply:
I could, but I wanted to keep the names as close as possible to the names used in the field descriptors definition: https://specs.frictionlessdata.io/table-schema/#field-descriptors